frame/contracts/src/exec.rs

```diff
@@ -903,11 +905,19 @@ where
 			// Every non delegate call or instantiate also optionally transfers the balance.
 			self.initial_transfer()?;
 
+			#[cfg(feature = "unsafe-debug")]
+			let code_hash = executable.code_hash().clone();
+			#[cfg(feature = "unsafe-debug")]
+			T::Debug::before_call(&code_hash, entry_point, &input_data);
```

Comment:
thought, maybe to keep things dry we could do something like this in pseudo code
```
let debug_call = T::Debug::start_call(...)
...
debug_call.end(&output)
```

Reply:
> this debug_call variable would have to be feature-gated as well, right?

yes so

```
#[cfg(feature = "unsafe-debug")]
let debug_call = T::Debug::start_call(...)
...
#[cfg(feature = "unsafe-debug")]
debug_call.end(&output)
```

to me more correct

Reply:
the problem I see with this approach is that we put more requirements for the user (and mock implementations) to implement another type for this `debug_call`